src/Text.Analyzers/Core/IdentifiersShouldBeSpelledCorrectly.cs

```diff
@@ -33,162 +43,465 @@ public abstract class IdentifiersShouldBeSpelledCorrectlyAnalyzer : DiagnosticAn
+
+                if (checkForUnmeaningful && symbolName.Length == 1)
+                {
+                    reportDiagnostic(GetUnmeaningfulIdentifierDiagnostic(symbol, symbolName));
+                }
+            }
+
+            IEnumerable<string> GetMisspelledWords(string symbolName)
+            {
+                var parser = new WordParser(symbolName, WordParserOptions.SplitCompoundWords);
+                if (parser.PeekWord() != null)
+                {
+                    var word = parser.NextWord();
+                    if (word == null)
+                    {
+                        yield break;
+                    }
+
+                    do
+                    {
+                        if (IsWordAcronym(word) || IsWordNumeric(word) || IsWordSpelledCorrectly(word))
+                        {
+                            continue;
+                        }
+
+                        yield return word;
+                    }
+                    while ((word = parser.NextWord()) != null);
+                }
+            }
+
+            static bool IsWordAcronym(string word) => word.All(char.IsUpper);
+
+            static bool IsWordNumeric(string word) => char.IsDigit(word[0]);
+
+            bool IsWordSpelledCorrectly(string word)
+                => !projectDictionary.UnrecognizedWords.Contains(word) && projectDictionary.RecognizedWords.Contains(word);
+        }
+
+        private static CodeAnalysisDictionary GetMainDictionary()
+        {
+            // The "main" dictionary, Dictionary.dic, was created in WSL Ubuntu with the following commands:
+            //
+            // Install dependencies:
+            // > sudo apt install hunspell-tools hunspell-en-us
+            // 
+            // Create dictionary:
+            // > unmunch /usr/share/hunspell/en_US.dic /usr/share/hunspell/en_US.aff > Dictionary.dic
+            //
+            // Tweak:
+            // Added the words: 'namespace'
+            var text = SourceText.From(TextAnalyzersResources.Dictionary);
+            return ParseDicDictionary(text);
+        }
+
+        private static CodeAnalysisDictionary ParseXmlDictionary(SourceText text)
+            => text.Parse(CodeAnalysisDictionary.CreateFromXml);
+
+        private static CodeAnalysisDictionary ParseDicDictionary(SourceText text)
+            => text.Parse(CodeAnalysisDictionary.CreateFromDic);
+
+        private static string RemovePrefixIfPresent(string prefix, string name)
+            => name.StartsWith(prefix, StringComparison.Ordinal) ? name[1..] : name;
+
+        private static Diagnostic GetMisspelledWordDiagnostic(ISymbol symbol, string misspelledWord)
+        {
+            return symbol.Kind switch
+            {
+                SymbolKind.Assembly => Diagnostic.Create(AssemblyRule, Location.None, misspelledWord, symbol.Name),
+                SymbolKind.Namespace => Diagnostic.Create(NamespaceRule, symbol.Locations.First(), misspelledWord, symbol.ToDisplayString()),
```

Comment:
```suggestion
                SymbolKind.Namespace => symbol.CreateDiagnostic(NamespaceRule, misspelledWord, symbol.ToDisplayString()),
```

Reply:
Are you suggesting to not include location?

Reply:
I am just suggesting to use extension method [CreateDiagnostic](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Utilities/Compiler/Extensions/DiagnosticExtensions.cs#L76) we have defined and already take care of reporting with the right location(s).

Reply:
Super cool; love it

Reply:
Done